department-of-veterans-affairs/va.gov-cms
# VACMS-11609: Update cypress definitions for see exist & visible

## Description

Closes #11609 
Related to #11611 

This PR creates a new definition for `I should see` so that it looks for an item visible on the page and not whether or not it exists (important for elements that are shown or hidden according to conditional logic, for example.)

`should be visible/not be visible` becomes `should see/not see`
`should be visible/not be visible` is removed
`should exist/not exist` replaces the former `should see/not see` definition

Tests that used any of the above are modified to reflect the changes.

## Bytes

Summary of work to now, from my analog notebook, transcribed for vis.

- [x] Create new `should exist` definition based on the old `I should see` > `should('exist')` one
- [x] Create a new `should ssee` for soft transition based on the old `should be visible` > `should('be.visible')` one
- [x] Delete the old `should see` definition when all tests pass
- [x] Rename `should ssee` to `should see`

Note: Below, **see > ssee** means change from `.should('exist')` to `.should('be.visible)` using the _new_ `I should see` definition.

- [x] content_editing_vamc_facility... passes:  **visible > ssee**
- [x] content_editing passes:
- - [x]  Sitewide alerts passes: **see > exist** 
- - [x]  Published not directly from node: **see > exist** 
- - [x] URL alias after node/taxonomy: no changes
- - [x] Press release country: **see > ssee** 
- - [x] Save and continue, revision: **not see > not exist** 
- - [x] Event form conditionals: **visible > ssee**
- - [x] Conditional form elements clear out: no changes
- [x] content_release passes: **see > exist**
- [x] entity_reference_validation passes: **see > exist**
- [x] ewa_block passes: no changes
- [x] file_upload passes: 
- [x] google_tag_manager passes: no changes
- [x] help_center passes:
- - [x] Get help link passes
- - [x] Access denied get help passes
- - [x] Not denied access don't get help passes
- [x] image_crop passes
- [x] media passes: **see > ssee** passes
- [x] navigation passes: **see > ssee** passes
- [x] password_policy passes
- [x] permissions passes: **see > ssee** 
- [x] step_definitions passes
- [x] text_validation passes
- [x] tooltip passes
- [x] user_login passes

## Testing done
Local cypress interactive testing. Search for replaced words and patterns.
Ran all tests locally.

## Screenshots
n/a

## QA steps
Tests need to pass with latest branch. Authors should check their tests to ensure that the definition change applies correctly. e.g. I should see vs should exist.

### Definition of Done


### Select Team for PR review



### Is this PR blocked by another PR?


### Does this PR need review from a Product Owner


### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
